pkg/profile/repo_test.go

```diff
@@ -0,0 +1,209 @@
+			Expect(err).To(MatchError(ContainSubstring("failed to parse profile")))
+			url, cloneBranch, _ := fakeGitClient.CloneArgsForCall(0)
+			Expect(url).To(Equal(repoURL))
+			Expect(cloneBranch).To(Equal(branch))
+		})
+	})
+	When("the profile.yaml empty", func() {
+		It("returns an error", func() {
+			profileYaml := []byte("")
+			path := "my-profile"
+			fakeGitClient.CloneStub = func(url string, branch string, dir string) error {
+				err := os.MkdirAll(filepath.Join(dir, path), 0755)
+				Expect(err).NotTo(HaveOccurred())
+				return ioutil.WriteFile(filepath.Join(dir, path, "profile.yaml"), profileYaml, 0755)
+			}
+			maker := profile.NewProfilesArtifactsMaker(profile.MakerConfig{
+				GitClient:        fakeGitClient,
+				RootDir:          "root-dir",
+				GitRepoNamespace: "git-repo-namespace",
+				GitRepoName:      "git-repo-name",
+			})
+			_, err := maker.GetProfileDefinition(repoURL, branch, "my-profile")
+			Expect(err).To(MatchError(ContainSubstring("failed to parse profile")))
+			url, cloneBranch, _ := fakeGitClient.CloneArgsForCall(0)
+			Expect(url).To(Equal(repoURL))
+			Expect(cloneBranch).To(Equal(branch))
+		})
+	})
+	Context("caching", func() {
+		var (
+			profileYaml []byte
+			path        string
+			maker       *profile.ProfilesArtifactsMaker
+		)
+		BeforeEach(func() {
+			profileYaml = []byte(`
+apiVersion: packages.weave.works.io/v1alpha1
+kind: Profile
+metadata:
+  name: nginx
+spec:
+  description: foo
+  artifacts:
+    - name: bar
+      kustomize:
+        path: baz`)
+			path = "my-profile"
+			fakeGitClient.CloneStub = func(url string, branch string, dir string) error {
+				err := os.MkdirAll(filepath.Join(dir, path), 0755)
+				Expect(err).NotTo(HaveOccurred())
+				return ioutil.WriteFile(filepath.Join(dir, path, "profile.yaml"), profileYaml, 0755)
+			}
+			maker = profile.NewProfilesArtifactsMaker(profile.MakerConfig{
+				GitClient:        fakeGitClient,
+				RootDir:          "root-dir",
+				GitRepoNamespace: "git-repo-namespace",
+				GitRepoName:      "git-repo-name",
+			})
+		})
+		When("called multiple times with the same repo", func() {
+			It("it only clones it once", func() {
+				_, err := maker.GetProfileDefinition(repoURL, branch, "my-profile")
+				Expect(err).NotTo(HaveOccurred())
+				_, err = maker.GetProfileDefinition(repoURL, branch, "my-profile")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fakeGitClient.CloneCallCount()).To(Equal(1))
+			})
+		})
+		When("called multiple times with different url", func() {
+			It("it clones all", func() {
+				_, err := maker.GetProfileDefinition(repoURL, branch, "my-profile")
+				Expect(err).NotTo(HaveOccurred())
+				_, err = maker.GetProfileDefinition("https://github.com/different", branch, "my-profile")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fakeGitClient.CloneCallCount()).To(Equal(2))
+			})
+		})
+		When("called multiple times with the same url but different branch", func() {
```

Comment:
these tests exist in `repo_test` and `artifact_test`?

Reply:
Yes, because different code path. One is the local checkout and the other is in the profiles cloner.